### PR TITLE
added crud operation for all entities, removed sensitive information from configs

### PR DESCRIPTION
**Pull request by alexkonkin**
-added crud operation for all entities
-updated docker-compose to support transactions
-removed sensitive information from configuration files

### CODE REVIEW

**Comment by TonyYoda** [comment]
@alexkonkin 
please move the logic from controller to the corresponding service, your code will be more readable.
Please let me know next time when you submit your PR

File: routes/products.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        let product = new Product({
+            name,
+            description,
+            price,
+            pictures,
+            category_id,
+            stock_quantity
+        });
+
+        product = await product.save();
```
Comment (TonyYoda) [inline routes/products.js]: Handle the case when category does not exists.

File: routes/products.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        logger.info('Deleting product', { id: req.params.id });
+
+        const reviewCount = await Review.countDocuments({ product_id: req.params.id }).session(session);
+        const orderCount = await Order.countDocuments({ product_id: req.params.id }).session(session);
+
+        if (reviewCount > 0 /*|| orderCount > 0*/) {
+            logger.info('Cannot delete product associated with existing reviews or orders', { id: req.params.id, reviewCount, orderCount });
+            await session.abortTransaction();
+            session.endSession();
+            return res.status(400).send('Cannot delete the product as it is associated with existing reviews or orders.');
```
Comment (TonyYoda) [inline routes/products.js:152]: In case product has associated reviews or orders product should be marked as deleted.

File: uncaughtExceptions.log
Comment (TonyYoda) [inline uncaughtExceptions.log:1]: Do not store logs file in the repo

**Review by TonyYoda — APPROVED** [review]

File: uncaughtExceptions.log
Comment (alexkonkin) [reply uncaughtExceptions.log:1]: removed log that was added by mistake via gitignore

File: routes/products.js (hunk 1 above)
Comment (alexkonkin) [reply routes/products.js]: if the category does not exist, the product should not be added

**Pull request merged**